components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAOTest.java

```diff
@@ -0,0 +1,1690 @@
+
+        /*
+            Unit Testing for 'getIdPsSearch' method with following parameters
+            getIdPsSearch(int tenantId, List<ExpressionNode> expressionNode, int limit, int offset, String sortOrder,
+            String sortBy)
+        */
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            List<IdentityProvider> idps = idPManagementDAO.getIdPsSearch(tenantId, expressionNodes, limit, offset,
+                    sortOrder, sortBy);
+            assertEquals(idps.size(), count);
+            if (count > 0) {
+                assertEquals(idps.get(0).getIdentityProviderName(), firstIdp);
+            }
+        }
+    }
+
+    @DataProvider
+    public Object[][] getIdPsSearchWithExpressionNodesExceptionData() {
+
+        ExpressionNode en1 = new ExpressionNode();
+        List<ExpressionNode> expressionNodes1 = new ArrayList<>();
+        expressionNodes1.add(en1);
+
+        ExpressionNode en2 = new ExpressionNode();
+        List<ExpressionNode> expressionNodes2 = new ArrayList<>();
+        en2.setAttributeValue("InvalidAttribute");
+        en2.setOperation("eq");
+        en2.setValue("testIdP1");
+        expressionNodes2.add(en2);
+
+        ExpressionNode en3 = new ExpressionNode();
+        List<ExpressionNode> expressionNodes3 = new ArrayList<>();
+        en3.setAttributeValue("description");
+        en3.setOperation("InvalidOperation");
+        en3.setValue("2");
+        expressionNodes3.add(en3);
+
+        return new Object[][]{
+                {SAMPLE_TENANT_ID, expressionNodes1, 2, 0, "WrongOrder", "NAME", "ServerException"},
+                {SAMPLE_TENANT_ID, expressionNodes1, 2, 0, "DESC", "WrongBy", "ServerException"},
+                {SAMPLE_TENANT_ID, expressionNodes2, 1, 0, "ASC", "NAME", "ClientException"},
+                {SAMPLE_TENANT_ID, expressionNodes3, 1, 0, "ASC", "NAME", "ClientException"},
+        };
+    }
+
+    @Test(dataProvider = "getIdPsSearchWithExpressionNodesExceptionData")
+    public void testGetIdPsSearchWithExpressionNodesException(int tenantId, List<ExpressionNode> expressionNodes,
+                                                              int limit, int offset, String sortOrder, String sortBy,
+                                                              String exceptionType) throws Exception {
+
+        /*
+            Unit Testing for 'getIdPsSearch' method with following parameters
+            getIdPsSearch(int tenantId, List<ExpressionNode> expressionNode, int limit, int offset, String sortOrder,
+                           String sortBy)
+        */
```

Comment:
Do we need this comment here?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/commit/6b030460eec87be9f79276256c9a3617fe357398